### PR TITLE
:sparkles: meaningful name for injected components

### LINKED ISSUE #7: Add a meaningful display name to injected components
Have a meaningful display name like :

```jsx
const Example = () => <div />
const Injected = inject()(Example)
// Injected.displayName == 'inject(Example)'
```

### PR DESCRIPTION
**Pull request by EmrysMyrddin**
fixes #7

### CODE REVIEW

File: src/inject.jsx [hunk 1]
```diff
@@ -2,7 +2,13 @@ import React from 'react'
 import { autorun, isObservable, toJS } from 'mobx'
 import shallowEqual from 'fbjs/lib/shallowEqual'
 
-export default injectFunction => Component => class extends React.Component {
+export default injectFunction => Component =>
```
Comment (fabienjuif) [inline src/inject.jsx]: What about this ?
```js
export default injectFunction => Component => {
  const getWrappedDisplayName = () => (
    Component.displayName 
    || Component.name 
    ||(Component.constructor && Component.constructor.name)
    || 'Unknown'
  )

  return class extends React.Component {
    static displayName = getWrappedDisplayName()
    // ...
```

Comment (EmrysMyrddin) [reply src/inject.jsx]: Yes I can, I was just trying to not add curlybraces to keep the nice currying style !

Comment (EmrysMyrddin) [reply src/inject.jsx]: But perhaps we don't need a function ?

Comment (fabienjuif) [reply src/inject.jsx]:
```js
const getWrappedDisplayName = Component => (
    Component.displayName 
    || Component.name 
    ||(Component.constructor && Component.constructor.name)
    || 'Unknown'
  )

export default injectFunction => Component => class extends React.Component {
  static displayName = getWrappedDisplayName(Component)
  // ...
}
```

Comment (fabienjuif) [reply src/inject.jsx]: This is a pure function implementation that keeps currying style

File: src/inject.jsx [hunk 2]
```diff
@@ -2,12 +2,15 @@ import React from 'react'
 import { autorun, isObservable, toJS } from 'mobx'
 import shallowEqual from 'fbjs/lib/shallowEqual'
 
-export default injectFunction => Component =>
-  class extends React.Component {
-    static displayName = `inject(${Component.displayName ||
-      Component.name ||
-      (Component.constructor && Component.constructor.name) ||
-      'Unknown'})`
+const getWrappedDisplayName = Component => `inject(${
```
Comment (fabienjuif) [inline src/inject.jsx]: either :
 - `getWrappedDisplayName`:
```js
getWrappedDisplayName = Component => (
Component.displayName || /* etc */
)
```

or
 - **getDisplayName**
```js
getWrappedDisplayName = Component => `inject(${
  Component.displayName || /* etc */
`)
```

File: src/inject.jsx [hunk 3]
```diff
@@ -2,7 +2,16 @@ import React from 'react'
...
 
+const getDisplayName = Component => `inject(${
+  Component.displayName
+  || Component.name
+  || (Component.constructor && Component.constructor.name)
+  || 'Unknown'
+})`
+
 export default injectFunction => Component => class extends React.Component {
+  static displayName = getWrappedDisplayName(Component)
```
Comment (fabienjuif) [inline src/inject.jsx]: 🤣

**Pull request merged**